src/poetry/core/packages/directory_dependency.py

```diff
@@ -55,10 +40,44 @@ def __init__(
             source_url=self._full_path.as_posix(),
             extras=extras,
         )
-
         # cache this function to avoid multiple IO reads and parsing
         self.supports_poetry = functools.lru_cache(maxsize=1)(self._supports_poetry)
 
+    def _validate(self) -> None:
```

Comment:
I think we should try the approach mentioned in https://github.com/python-poetry/poetry-core/pull/210#issuecomment-1159010010. We need two modes for `_validate`: a warning mode and an exception mode. The warning mode can be used in the `__init__` and the exception mode in `full_path`. I think it should be save to assume that when accessing `full_path` the directory must exist (and be a valid project).